osu.Game/Skinning/Editor/SkinEditorOverlay.cs

```diff
@@ -73,9 +73,13 @@ public override void Show()
                 skinEditor = new SkinEditor(target);
                 skinEditor.State.BindValueChanged(editorVisibilityChanged);
 
-                Debug.Assert(skinEditor != null);
-
-                LoadComponentAsync(skinEditor, AddInternal);
+                // Schedule ensures that if `Show` is called before this overlay is loaded,
+                // it will not throw (LoadComponentAsync requires the load target to be in a loaded state).
+                Schedule(() =>
```

Comment:
I can see a few issues with this:

* If `Show()` is called several times before load complete (or, less likely, multiple times within a frame), multiple instances will load and get added.
* This delegate uses an implicit `this` capture, so the `skinEditor` from the outer scope is not necessarily the same `skinEditor` in the inner scope:

![2021-12-29-132508_945x259_scrot](https://user-images.githubusercontent.com/20418176/147661972-5ba20ae0-d151-40bb-983e-4e7a277179c2.png)

Reply:
Am I missing something? I left the instance creation outside the schedule for this reason.

Reply:
`skinEditor` is a class field, so I think the compiler figures that it may as well capture `this` rather than also capture the concrete value of that field at delegate construction time. Copying out to a variable looks to prevent this:

![2021-12-29-133002_965x466_scrot](https://user-images.githubusercontent.com/20418176/147662269-5f02738a-ed0d-40c9-a1be-83648ae4073a.png)

Reply:
Thoroughly confused by what you're saying. Can you re-read the code with special attention to the `skinEditor` null check and confirm you can see an issue? I still cannot.

Reply:
Hmm, I thoroughly missed the fact that the null checks all over were trying to enforce that only one instance of the editor is ever created. *Buuuut* there's still `SkinEditorOverlay.Reset()` which potentially breaks this.

So it'd have to be something like
```
skinEditorOverlay.Show();
skinEditorOverlay.Reset();
skinEditorOverlay.Show();
```
all called before the overlay has loaded. Would schedule two delegates that would access the new value of `skinEditor`, therefore ending up trying to double-add the instance created with the second `Show()` call. Unsure if a big deal or not...

Reply:
Yeah, I did see that one. Probably worth fixing.

Reply:
Extracting the variable in b1a4441 fixes the potential crash by a double-add in the above scenario, but it doesn't fix the possibility of a double-add itself occuring. For that to be the case `Reset()` would have to cancel this delegate. Or you could `AddOnce()` this and call it a day?

Reply:
Not sure it matter as `Expire` would have been called on the previous instance. Can change if you think it's worth it.

Reply:
I suppose so. Probably fine as is then.

Reply:
I refactored things anyway, I think it reads better now.